lib/vector-core/src/event/vrl_target.rs

```diff
@@ -24,6 +24,7 @@ pub enum VrlTarget {
     // that `fields` must always be a `Map` variant.
     LogEvent(Value, EventMetadata),
     Metric(Metric),
+    Trace(Value, EventMetadata),
```

Comment:
I'd be tempted to leave traces out of VRL for now, mostly because I'm not sure if we'll want to give them the same API as logs or do something more specific. One of the goals with the initial implementation is to limit the surface area so that it's easier to change as we get more experience with traces.

Reply:
I see one issue there, one of the major use case for traces is demultiplexing a trace flow based on various criterion, thus using the `filter`/`route` transforms, but we would need to be able to evaluate VRL condition. I don't see that as a blocker, as the whole VRL with trace thing need to be carefully though. So I would say it sounds reasonable, at the moment, to let traces out of VRL and plan that a bit later to gain a wider consensus.

Reply:
Agreed on this. I'd rather we write an RFC first to think through the interaction model. We already have half-baked support for metrics that I'd wish we had left out until we're ready to fully commit to it, this seems like more baggage we're not ready to carry yet, until some thinking has been done on this.

Reply:
I see several alternatives:
1. We can keep VRL support for trace, but disable it at the topology level for the `remap` transform, letting only the VRL condition for `filter`/`route` transforms and preventing trace manipulation in VRL
2. We let VRL enabled including the `remap` transform
3. We let traces completely out of VRL

Users will probably expect to get at least `filter` & `route` to work, I'm not so sure about the rest.